tests/savegame_test.cpp

```diff
@@ -0,0 +1,268 @@
+        {{253, 87, 0},{ "mon_zombie", {136, 25, 0}}},
+        {{259, 95, 0},{ "mon_zombie", {143, 122, 0}}},
+        {{259, 94, 0},{ "mon_zombie", {139, 109, 0}}},
+        {{259, 91, 0},{ "mon_dog", {139, 82, 0}}},
+        {{194, 87, -3},{"mon_irradiated_wanderer_4", {119, 73, -3}}},
+        {{194, 87, -3},{ "mon_charred_nightmare", {117, 83, -3}}},
+        {{142, 96, 0},{ "mon_deer", {16, 109, 0}}},
+        {{196, 66, -1},{ "mon_turret", {17, 65, -1}}},
+        {{196, 63, -1},{ "mon_broken_cyborg", {19, 26, -1}}}
+    };
+    for( auto candidate_monster : expected_monsters ) {
+        REQUIRE(test_map.monster_check(candidate_monster));
+    }
+    // Check NPCs.  They're complicated enough that I'm just going to spot-check some stats.
+    for( const npc *test_npc : test_map.npcs ) {
+        if( test_npc->disp_name() == "Felix Brandon" ) {
+            REQUIRE(test_npc->get_str() == 7);
+            REQUIRE(test_npc->get_dex() == 8);
+            REQUIRE(test_npc->get_int() == 7);
+            REQUIRE(test_npc->get_per() == 10);
+            REQUIRE(test_npc->get_skill_level("barter") == 4);
+            REQUIRE(test_npc->get_skill_level("driving") == 2);
+            REQUIRE(test_npc->get_skill_level("firstaid") == 7);
+            REQUIRE(test_npc->get_skill_level("mechanics") == 5);
+            REQUIRE(test_npc->get_skill_level("dodge") == 3);
+            REQUIRE(test_npc->get_skill_level("launcher") == 3);
+            REQUIRE(test_npc->pos() == tripoint(168, 66, 0));
+        } else if( test_npc->disp_name() == "Mariann Araujo" ) {
+            REQUIRE(test_npc->get_str() == 11);
+            REQUIRE(test_npc->get_dex() == 9);
+            REQUIRE(test_npc->get_int() == 10);
+            REQUIRE(test_npc->get_per() == 10);
+            REQUIRE(test_npc->get_skill_level("barter") == 4);
+            REQUIRE(test_npc->get_skill_level("driving") == 0);
+            REQUIRE(test_npc->get_skill_level("firstaid") == 5);
+            REQUIRE(test_npc->get_skill_level("bashing") == 5);
+            REQUIRE(test_npc->get_skill_level("dodge") == 4);
+            REQUIRE(test_npc->pos() == tripoint(72, 54, 0));
+        } else {
+            // Unrecognized NPC, fail.
+            REQUIRE(false);
+        }
+    }
+}
+
+TEST_CASE("Reading a legacy overmap save.") {
+
+    init_global_game_state();
+
+    overmap test_map;
+    std::ifstream fin;
+
+    fin.open("tests/data/legacy_0.C_overmap.sav");
+    if( fin.is_open() ) {
+        test_map.unserialize(fin);
+        fin.close();
+    } else {
+        REQUIRE(false);
+    }
+    check_test_overmap_data(test_map);
+
+    std::ofstream fout;
+
+    fout.open("tests/data/jsionized_overmap.sav");
+    if( fout.is_open() ) {
+        test_map.serialize(fout);
+        fout.close();
+    }
+
+    overmap test_map_2;
+
+    fin.open("tests/data/jsionized_overmap.sav");
+    if( fin.is_open() ) {
+        test_map_2.unserialize(fin);
+        fin.close();
+    } else {
+        REQUIRE(false);
```

Comment:
Because I didn't think of it, and it's not technically a REQUIRE since that implies the thing you're testing is failing instead of the test itself, but it's convinient and cleans things up, so I think I'll change it to that.